syntax = "proto3";

package userservice;

service UserService {
    rpc AddUser (UserRequest) returns (UserResponse);
    rpc GetUser (UserIdRequest) returns (UserResponse);
    rpc UpdateUser (UserRequest) returns (UserResponse);
}

message UserRequest {
    string name = 1;
    string email = 2;
    int32 age = 3;
}

message UserResponse {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}

message UserIdRequest {
    int32 id = 1;
}
